0002-crypto-conditions/0002-crypto-conditions.md

```diff
@@ -110,508 +115,638 @@ informative:
+It must be possible to verify that all types used in a crypto-condition are supported (including the types and subtypes of any sub-crypto-conditions) even if the fulfillment is not available to be analysed yet. Therefore, all compound conditions set the bits in this bitmap that correspond to the set of types and subtypes of all sub-crypto-conditions.
 
-A prefix condition prepends the message to be validated with a constant string before passing it on to the subcondition's validation function. <!-- TODO: better explanation of why -->
+The field is encoded as a variable length BIT STRING, as defined in ASN.1 to accommodate new types that may be defined. 
 
-## THRESHOLD {#threshold-feature-suite}
-The THRESHOLD feature suite provides conditions that have several weighted subconditions and a threshold number. This feature suite is assigned the feature bit 2^3 = 0x08.
+Each bit in the bitmap represents a type from the list of known types in the IANA-maintained [Crypto-Condition Type Registry](#crypto-conditions-type-registry) and the bit corresponding to each type is the bit at position X where X is the type ID of the type. 
 
-Threshold conditions provide flexible multi-signing, such as requiring "M-of-N" subconditions be fulfilled. Subconditions can also be weighted so that one subcondition can count multiple times towards meeting the threshold.
+The presence of one or more sub-crypto-conditions of a specific type is indicated by setting the numbered bit corresponding to the type ID of that type.
 
-## RSA-PSS {#rsa-pss-feature-suite}
-The RSA-PSS feature suite provides the RSS-PSA signature algorithm. RSA-PSS is a signature algorithm based on the RSA cryptosystem, which relates to the problem of factoring the product of two large prime numbers. This feature suite is assigned the feature bit 2^4 = 0x10.
+For example, a compound condition that contains an ED25519 crypto-condition as a sub-crypto-condition will set the bit at position 4. 
 
-## ED25519 {#ed25519-feature-suite}
-The ED25519 feature suite provides the Ed25519 signature algorithm. Ed25519 is a signature algorithm based on the EdDSA signing scheme and the compact elliptic curve known as Ed25519. This feature suite is assigned the feature bit 2^5 = 0x20.
+Bits are numbered per the encoding rules for the ASN.1 BIT STRING type.
 
+## Fulfillment {#fulfillment-format}
 
+The ASN.1 definition for fulfillments is defined as follows:
 
+    Fulfillment ::= CHOICE {
+	  preimage PreimageFulfillment ,
+	  prefix PrefixFulfillment,
+      threshold ThresholdFulfillment,
+	  rsaSha256 RsaSha256Fulfillment,
+	  ed25519 Ed25519Fulfillment
+    }
 
-# Condition Types
-The following condition types are defined in this version of the specification. Future versions of this spec MAY introduce new feature suites and condition types, which SHALL be registered in the IANA maintained [Crypto-Condition Type Registry](#crypto-conditions-type-registry).
+    PreimageFulfillment ::= OCTET STRING
```

Comment:
I would really like to ask to keep every fulfillment a SEQUENCE. It makes it a lot harder to encode this as just an octet string instead of a SEQUENCE with a single element.